### DISCUSSION TITLE
How to list all tables in Prisma

### DISCUSSION THREAD

**Discussion by bastianhilton** [prisma/prisma#17125] [Q&A]
Hello i'm using Prisma with Nuxt 3 and i'm building a database application that requires me to list all the tables from a Prisma Schema. 

it would be something like this but its not working:

```
import { defineEventHandler } from 'h3';
import { usePrisma } from '@sidebase/nuxt-prisma'

export default defineEventHandler(async (event) => {
  const prisma = usePrisma(event)

  return prisma.$queryRaw`SELECT name FROM public WHERE type='table'`
})
```

Any idea how to do this?

**Comment by nurul3101** [comment]
Hello @bastianhilton 👋 

Here's a working example for listing table names for a database. (Assuming a PostgreSQL database)

```typescript
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

// A `main` function so that you can use async/await
async function main() {
  // ... you will write your Prisma Client queries here

  const tableNames = await prisma.$queryRaw`SELECT table_name
  FROM information_schema.tables
  WHERE table_schema = 'postgres'
  `;

  console.log(tableNames);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

Here's the response:
<img width="494" alt="Screenshot 2023-01-04 at 11 55 50 AM" src="https://user-images.githubusercontent.com/13049676/210496404-98d9b306-ef45-46e8-870d-81bde25ef464.png">

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
